File: qqbot-module-onebot11/src/main/java/com/mieai/qqbot/onebot11/transport/OneBotTransportStatus.java
package com.mieai.qqbot.onebot11.transport;

public record OneBotTransportStatus(
        String state,
        boolean forwardListening,
        int forwardConnections,
        boolean reverseConnected,
        String lastError) {

    public static OneBotTransportStatus disabled() {
        return new OneBotTransportStatus("DISABLED", false, 0, false, null);
    }

    public static OneBotTransportStatus waiting() {
        return new OneBotTransportStatus("WAITING_FOR_BOT", false, 0, false, null);
    }

    public static OneBotTransportStatus failed(String error) {
        return new OneBotTransportStatus("FAILED", false, 0, false, error);
    }
}

File: qqbot-module-onebot11/src/main/kotlin/com/mieai/qqbot/onebot11/transport/OneBotTransportStatus.kt
package com.mieai.qqbot.onebot11.transport

data class OneBotTransportStatus(
    val state: String,
    val forwardListening: Boolean,
    val forwardConnections: Int,
    val reverseConnected: Boolean,
    val lastError: String?,
) {
    companion object {
        fun disabled() = OneBotTransportStatus("DISABLED", false, 0, false, null)

        fun waiting() = OneBotTransportStatus("WAITING_FOR_BOT", false, 0, false, null)

        fun failed(error: String) = OneBotTransportStatus("FAILED", false, 0, false, error)
    }
}
